athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -0,0 +1,471 @@
+     * @param request   Provides details on who made the request and which Athena catalog, database, and table they are querying.
+     * @return A GetTableResponse which primarily contains:
+     * 1. An Apache Arrow Schema object describing the table's columns, types, and descriptions.
+     * 2. A Set<String> of partition column names (or empty if the table isn't partitioned).
+     * 3. A TableName object confirming the schema and table name the response is for.
+     * 4. A catalog name corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public GetTableResponse doGetTable(BlockAllocator allocator, GetTableRequest request)
+    {
+        TableName tableInfo = request.getTableName();
+        LOGGER.debug("MetadataHandler=GcsMetadataHandler|Method=doGetTable|Message=queryId {}",
+                request.getQueryId());
+        if (this.datasource == null) {
+            throw new RuntimeException("Metastore couldn't be initialized");
+        }
+        LOGGER.debug("MetadataHandler=GcsMetadataHandler|Method=doGetTable|Message=Schema name {}, table name {}",
+                tableInfo.getSchemaName(), tableInfo.getTableName());
+        datasource.loadAllTables(tableInfo.getSchemaName());
+        LOGGER.debug(MessageFormat.format("Running doGetTable for table {0}, in schema {1} ",
+                tableInfo.getTableName(), tableInfo.getSchemaName()));
+        Schema schema = gcsSchemaUtils.buildTableSchema(this.datasource,
+                tableInfo.getSchemaName(),
+                tableInfo.getTableName());
+        Schema partitionSchema = getPartitionSchema();
+        return new GetTableResponse(request.getCatalogName(), request.getTableName(), schema,
+                partitionSchema.getFields().stream().map(Field::getName).collect(Collectors.toSet()));
+    }
+
+    /**
+     * Used to get the partitions that must be read from the request table in order to satisfy the requested predicate.
+     *
+     * @param blockWriter        Used to write rows (partitions) into the Apache Arrow response.
+     * @param request            Provides details of the catalog, database, and table being queried as well as any filter predicate.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing work for a query that has already terminated
+     */
+    @Override
+    public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request, QueryStatusChecker queryStatusChecker)
+    {
+        LOGGER.debug("RecordHandler=GcsMetadataHandler|Method=getPartitions|Message=queryId {}", request.getQueryId());
+        LOGGER.debug("readWithConstraint: schema[{}] tableName[{}]", request.getSchema(), request.getTableName());
+
+        TableName tableName = request.getTableName();
+        if (this.datasource == null) {
+            throw new RuntimeException("Table " + tableName.getTableName() + " not found in schema "
+                    + tableName.getSchemaName());
+        }
```

Comment:
exception doesn't match condition

Reply:
Code removed
The changes are made in a commit eb79bce67883879420769420f4ed04a83cf8bbc6